site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -57,20 +57,16 @@ public function openQueue() {
             );
         }
 
-        if (empty($_POST['token'])) {
-            $this->core->addErrorMessage("Missing secret code");
-            return MultiResponse::RedirectOnlyResponse(
-                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
-            );
-        }
-
         //Replace whitespace with "_"
         $queue_code = trim($_POST['code']);
-        $token = trim($_POST['token']);
+        $token = empty($_POST['token']) ? '' : trim($_POST['token']);
```

Comment:
I think `isset` would be the better check here since it only checks to make sure a value is set, whereas `empty` checks for a value that is truthy to empty (0,false,empty string).

Because of that the value `0` cannot be used as a secret